src/Network/Pusher/Data.hs

```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE RecordWildCards #-}
```

Comment:
Actually I think these should be a record instead of a `HashSet`. The reason is it gives you more type safety; i.e. you know at compile time what the fields are, so you can look up the values and know the exists. With a `HashSet` you have to handle potential `Nothing` values whenever you attempt to lookup a value.

As for which approach to go for with the records, I don't feel too strongly. My personal preference would actually be the first, but I'll leave it up to you. Bear in mind you can just do `toJSON apns =` instead of `toJSON apns@(Apns _ _ _ _ _ _) =`.